### PR TITLE
backend: create unified BTC/LTC accounts, with option for splitting

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: backend/backend.go [hunk 1]
```diff
@@ -320,11 +428,11 @@ func (backend *Backend) createAndAddAccount(
 	prefix := "eth-erc20-"
 	if strings.HasPrefix(code, prefix) {
 		if !backend.config.AppConfig().Backend.ETH.ERC20TokenActive(code[len(prefix):]) {
-			log.WithField("name", name).Info("skipping inactive erc20 token")
```
Comment (benma) [inline backend/backend.go:323]: field already present, see L427

File: backend/backend.go [hunk 2]
```diff
@@ -309,6 +309,111 @@ func (backend *Backend) CreateAndAddAccount(
...
+		), nil
+	}
+
+	splitAccounts := backend.config.AppConfig().Backend.SplitAccounts ||
+		!keystore.SupportsUnifiedAccounts()
+	if splitAccounts {
+		for _, cfg := range supportedConfigs {
+			cfg := cfg
+			getSigningConfigurations := func() (signing.Configurations, error) {
+				signingConfiguration, err := getSigningConfiguration(cfg)
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/backend.go:374]: can only be fixed by even longer names:

`getMultipleSigningConfigurations`

:D just kidding

File: backend/backend.go [hunk 3]
```diff
@@ -309,6 +309,111 @@ func (backend *Backend) CreateAndAddAccount(
...
+	}
+}
+
+// adds a combined BTC account with the given script types. If the keystore requires split accounts
+// (bitbox01) or the user configure split accounts in the settings, one account per script type is
+// added instead of a combined account.
+func (backend *Backend) createAndAddBTCAccount(
+	keystore keystore.Keystore,
+	coin coin.Coin,
+	code string,
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/backend.go:334]: that would be available already in `coin.Code()`. this is actually the account code. which can be different to the coin code (e.g. in the three split accounts they have to have different codes).

should make a type for this too some day.

File: backend/backend.go [hunk 4]
```diff
@@ -309,6 +309,111 @@ func (backend *Backend) CreateAndAddAccount(
...
+			switch cfg.scriptType {
+			case signing.ScriptTypeP2PKH:
+				nameSuffix = ": legacy"
+			case signing.ScriptTypeP2WPKH:
+				nameSuffix = ": bech32"
+			}
+			err := backend.CreateAndAddAccount(
+				coin,
+				fmt.Sprintf("%s-%s", code, cfg.scriptType),
+				name+nameSuffix,
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/backend.go:390]: First one doesn't work for when there is no suffix. went with 2nd one.

File: backend/backend.go [hunk 5]
```diff
@@ -309,6 +309,111 @@ func (backend *Backend) CreateAndAddAccount(
...
+	code string,
+	name string,
+	configs []scriptTypeWithKeypath,
+) {
+	log := backend.log.WithField("code", code).WithField("name", name)
+	if !backend.config.AppConfig().Backend.CoinActive(coin.Code()) {
+		log.Info("skipping inactive account")
+		return
+	}
+	supportedConfigs := []scriptTypeWithKeypath{}
```
[the remark this answers is not in the document]
Comment (benma) [reply backend/backend.go:343]: done

File: backend/backend.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (x1ddos) [reply backend/backend.go:374]: :see_no_evil:

**Pull request merged**